stm32: clock_control: Driver overhaul

erwango opened the pull request:
In order to driver future evolutions (mainly support for alternate clocks, cf https://github.com/zephyrproject-rtos/zephyr/pull/42097), revise stm32 clock_control driver implementation.
This change is adapted from what was done initially on H7 driver (cf https://github.com/zephyrproject-rtos/zephyr/pull/41585) and extended to other series (common and U5 clock_control drivers). Only MP1 is left aside for now

Mainly, two changes are done:
- Break down sysclock initialization in elementary steps easily configurable by device tree
- Introduce new bus clock bindings which simplify _on and _off implementation and will be (easily ?) reused for reset control

Based on #44291, which allow to verify no regression is introduced.

erwango committed:
dts/arm: stm32h7: Introduce a stm32h7 specific clock binding

This new binding allows to work on providing stm32h7 specific
alternate and complementary device clocks.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32h7: Change clock bus bindings values

Set bus binding values using registers offset values.
As a consequence update driver to take this into account
in clock_on and clock_off functions.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
include/drivers: stm32_clock_control: Avoid non defined PLL outputs

On some series (H7, U5), it is possible define clock configuration
with disabled PLL outputs.
In that case, it is legit that matching pll property is not available.
Define corresponding STM32_PLLX_Y_DIVISOR macros using DT_PROP_OR
to avoid build issues in case prop is not available.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
include/dt-bindings/clock: Add stm32u5 clock bindings

Introduce a new scheme to define clock bingings on u5.
In a next steps, this new scheme will allow to provide u5 specific
alternate and complementary device clocks.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32h7: Fixes on function declarations

Fix minor issues on some functions headers.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32u5: Use new clock bindings

This change updates stm32u5 driver to make use of new clock bindings.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32u5: Update SystemCoreClock at a single place

Move update of CMSIS variable SystemCoreClock at a single place
in a more direct way.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32u5: Factorize bus prescalers settings

Move prescaler settings to the clock_control_init function.
At this step they will be set up twice in PLL case, this will
be fixed in a next step.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32u5: Don't disable other clocks

Don't disable other clocks after a clock is configured.
This should be left to the API.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32u5: Centralize flash_latency update

Flash latency setting could be factorized in a single location,
rather than split in each clock setting function.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32u5: Centralize regu voltage setting

Similar to other general settings, centralize regu voltage
setting.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32u5: Move clock init to a single function

Move fixed clocks initialization to a single function.
Benefit is they could now be enabled independently of the
main clock configuration based on dts status and then be
used by peripherals even is not part of the main clock tree.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32u5: Clean up fixed clocks functions.

Now that fixed clocks are enabled in a single function, a
bunch of functions could now be removed.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32u5: Use fixed clocks set up to configure PLLs

Take advantage of previous work to configure PLL and remove
usage of LL_PLL1_ConfigSystemClock_FOO utils functions.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32 common: Don't disable fixed clocks

Each clock should be configured individually by device tree,
don't disable them blindly.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32 common: Factorize freq bus setting

Factorize setting of frequency for busses.
Additionally, factorize SysCoreClock update.

The operations are now done twice in case of PLL since they are part
of LL utils PLL configuration function, but they are removed in next
commits.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32 common: Factorize flash latency setting

Make use of LL_SetFlashLatency in all cases to update flash latency.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32 common: Use DT macros for bus prescalers

Simplify and clean up driver code using STM32 clocks DT based macros.

Added STM32_FLASH_PRESCALER macro for this purpose.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32 common: Group fixed clocks init

Group fixed clocks inits in a unique set_up function.
Each clock is initialized depending on its dts status.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32_common: Add elementary PLL configuration step

Introduce a set_up_pll configuration function and make PLL configuration
an elementary step of the whole system clock configuration.

To implement this new, function make use of the existing series specific
files which allows series specific configuration when required.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
include/drivers/clock_control: stm32: Some HSE clocks are pure fixed-clocks

On STM32WB HSE clock is defined with "fixed-clock" compatible.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts: stm32: Add rcc prop undershoot-prevention

On some parts,  it could be required to use steps before applying
highest frequencies.
This was previously done as part of LL_PLL_ConfigSystemClock_FOO
utility functions which are no more used.
Use device tree to mention when this is required and implement it
in stm32_clock_control_init().

Additionally, fix the calls tp LL_RCC_SetAHBPrescaler, which require
use of ahb_prescaler helper.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32: style edits on set_up_fixed_clock_sources()

Review code style in set_up_fixed_clock_sources() for better
readability.
Use of 'if (IS_ENABLED(STM32_MSI_ENABLED))' inside '#if STM32_MSI_ENABLED'
is redundant but intentional as it is in line with remaining part of the
function (HSE/HSI cases).

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
include/drivers/clock_control: stm32: Missing PLL macro on some compats

STM32_PLL_ENABLED symbol is missing on F0/F1 and L0/L1 compatibles.
Fix that.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
include/dt-bindings/clock: stm32: Add bindings for 'common' series

Add new scheme clock bindings for 'common' series:
- stm32f1_clock.h > compatible with f0/f1/f3 series
- stm32f4_clock.h > compatible with f2/f4/f7 series
- stm32l0_clock.h > compatible with l0 series
- stm32l1_clock.h > compatible with l1 series
- stm32l4_clock.h > compatible with g4/l4/l5/wb series
- stm32wl_clock.h > compatible with wl series

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32 common: Use new bus clock bindings

Make use of new bus clocks bindings and make subsequent code
simplifications.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

alexanderwachter commented on drivers/clock_control/clock_stm32_ll_h7.c:
```diff
@@ -308,103 +308,47 @@ static uint32_t get_vco_output_range(uint32_t vco_input_range)
+
+	reg = (uint32_t *)(DT_REG_ADDR(DT_NODELABEL(rcc)) + pclken->bus);
+	reg_val = *reg;
+	reg_val &= ~pclken->enr;
```
```suggestion

```

erwango replied:
Good catch. Fixed and reported in stm32u5 and stm32 common drivers.

erwango committed:
boards: lora_e5_dev_board: Fix dts clock description

According to board documentation: "By default System
clock is driven by the MSI clock at 48MHz."

This is in line with rcc node dts configuration:
&rcc {
        [...]
	clocks = <&clk_msi>;
        [...]
};

Though pll node is currently enabled, which is not in line with
current dts clocks description scheme and results to compilation
issue in clock_control driver.
Remove pll node configuration to fix this.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts/bindings/clocks: st,stm32f105-pll-clock.yaml: previd is required

"prediv" property should be required and explicitly set as part of board
clock configuration.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32 common: Set flash latency code under switch

Some specific F1 variants don't handle flash latency.
Put flash latency dealing code under dedicated switch.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -114,6 +114,19 @@ static uint32_t get_pllsrc_frequency(void)
+		return STM32_HSI_FREQ;
+	default:
+		__ASSERT(0, "Unexpected startup freq");
+		return -EIO;
```
function `get_startup_frequency` is returning **uint32**

erwango replied:
fixed

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -226,152 +197,76 @@ static void set_regu_voltage(uint32_t hclk_freq)
-
-	/* Wait till MSIS is ready */
-	while (LL_RCC_MSIS_IsReady() != 1) {
+	r = get_vco_input_range(STM32_PLL_M_DIVISOR, &vco_input_range);
```
is it equivalent to `vco_input_range = get_vco_input_range(STM32_PLL_M_DIVISOR)`

erwango replied:
This way allows to generate errors on range issue.
For that part I'm fully reusing the existing code.

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -226,152 +197,76 @@ static void set_regu_voltage(uint32_t hclk_freq)
-	clock_switch_to_hsi(LL_RCC_SYSCLK_DIV_1);
-	LL_RCC_PLL1_Disable();
+	/* FRACN disable DIVP,DIVQ,DIVR enable*/
+	LL_RCC_PLL1FRACN_Disable();
```
changing the PLL1 config bits requires the PLL to be disabled, 
the LL_RCC_PLL1FRACN_Disable() does not guarantee that  _PLL1ON = 0 and PLL1RDY = 0_
This is done by  the `LL_RCC_PLL1_Disable`();   and evenmore with a check on `LL_RCC_PLL1_IsReady`

erwango replied:
`LL_RCC_PLL1_Disable()` is called at the beginning of this function.
`LL_RCC_PLL1FRACN_Disable` here is only meant to disable FRACN and reused from existing code (could be optionally cleaned up).
But I agree that comment is misleading, will fix.

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -226,152 +197,76 @@ static void set_regu_voltage(uint32_t hclk_freq)
+#endif /* STM32_PLL_ENABLED */
 
-#endif /* STM32_SYSCLK_SRC_PLL */
+	return 0;
```
` return; `
--> void config_src_sysclk_pll

erwango replied:
I think the change on this file is too big and the diff is misleading. 
Function `config_src_sysclk_pll` has been removed

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -386,149 +281,170 @@ void config_src_sysclk_hse(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
 
-	set_regu_voltage(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+	if (IS_ENABLED(STM32_MSIK_ENABLED)) {
+		/* Set MSIS Range */
```
/* Set MSI**K** Range */

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -386,149 +281,170 @@ void config_src_sysclk_hse(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
+		LL_RCC_LSI_Enable();
+		while (LL_RCC_LSI_IsReady() != 1) {
+		}
+	}
```
open for further update to include the LSI prediv  on the stm32u5 ( LL_RCC_LSI_SetPrescaler)

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -386,149 +281,170 @@ void config_src_sysclk_hse(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
+	/* Set voltage regulator to comply with targeted system frequency */
+	set_regu_voltage(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
 
-	ARG_UNUSED(dev);
```
still applicable

erwango replied:
It's not removed, but moved few lines above.
(well, actually not, but this is how github display changes when adding squashing all the changes, this should be clearer in individual commits)

erwango commented on drivers/clock_control/clock_stm32_ll_h7.c:
```diff
@@ -314,97 +314,42 @@ static inline int stm32_clock_control_on(const struct device *dev,
 	z_stm32_hsem_unlock(CFG_HW_RCC_SEMID);
 
-	return rc;
+	return 0;
```
ok, it is back now

erwango committed:
tests/drivers/clock_control: stm32f1: HSI clock is 8MHz

On STM32F series, HSI clock is 8MHz, fix test using 16MHz
and a test name.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
tests/drivers/clock_control: stm32l0/l1: MSI range 11 is not allowed

Remove L0 and L1 targets from "sysclksrc_msi_48" test case as this
MSI range 11 is not an allowed value on these series.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts/bindings/clocks: stm32l0-msi-clock: Use enum for allowed values

Use enum to describe the range of allowed MSI values.
This will help to detect configuration issues earlier.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented on drivers/clock_control/clock_stm32f1.c:
```diff
@@ -49,46 +44,61 @@ void config_pll_init(LL_UTILS_PLLInitTypeDef *pllinit)
-}
+	/* Configure PLL source */
+	if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
+		pll_source = LL_RCC_PLLSOURCE_HSI_DIV_2;
```
Right, will arrange code in order not to define it in case of PLL_SRC_HSI

erwango commented on drivers/clock_control/clock_stm32l4_l5_wb_wl.c:
```diff
@@ -27,19 +27,39 @@
+	uint32_t pll_source, pll_m, pll_n, pll_r;
+
 #ifdef PWR_CR5_R1MODE
 	/* set power boost mode for sys clock greater than 80MHz */
```
According to ref man it is more related to power and dynamic voltage scaling

FRASTM replied:
the question was more to catch where the `undershoot-prevention` PROPerty was handled in the clock driver ?

erwango replied:
This is done in following commit: https://github.com/zephyrproject-rtos/zephyr/pull/44312/commits/568b0aa9169ad076f7f37f2cca32bfe1033c59b6

FRASTM reviewed (commented):
Is the stm32f1 really different from the stm32f0  or stm32f3 : except the prediv which is fixed to 2 in case of pll source HSE, could both config_pll_sysclock functions clock config be factorized ?

erwango commented:
> Is the stm32f1 really different from the stm32f0 or stm32f3 : except the prediv which is fixed to 2 in case of pll source HSE, could both config_pll_sysclock functions clock config be factorized ?

There's actually one part of the F1 (!SOC_STM32F10X_DENSITY_DEVICE) which is compatible with one part of the F0/F3 (!RCC_PLLSRC_PREDIV1_SUPPORT). So putting them together will allow to factorize those, but we'll need to make 3 different configuration live in one single file. I don't see it as much more simple in the end.

FRASTM commented on drivers/clock_control/clock_stm32_ll_common.c:
```diff
@@ -386,64 +386,66 @@ static int set_up_plls(void)
+			/* Enable MSI */
+			LL_RCC_MSI_Enable();
+			while (LL_RCC_MSI_IsReady() != 1) {
+				/* Wait for HSI ready */
```
Wait for **M**SI ready

FRASTM commented on include/dt-bindings/clock/stm32l1_clock.h:
```diff
@@ -0,0 +1,19 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_STM32L0_CLOCK_H_
```
ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_STM32L**1**_CLOCK_H_

erwango committed:
dts/bindings/clock: st-stm32(wb)-rcc: Clarify 'clock-frequency' meaning

Rework bindings documentation to clearly illustrate the role of ahb
(and cpu1) prescaler which defines the actual core clock frequency,
and not only a bus frequency.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32 common: Remove intermediate hclk variable

CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC is the actual hclk freq (ie core clock);
Remove use of intermediate new_hclk_freq to fix and simplify code.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ABOSTM commented on dts/bindings/clock/st,stm32l0-msi-clock.yaml:
```diff
@@ -17,10 +17,11 @@ properties:
-        5: range 5 around 2.097 MHz (reset value)
-        6: range 6 around 4.194 MHz
+      enum:
+        - 0 # range 0 around 100 kHz
+        - 1 # range 1 around 200 kHz
+        - 2 # range 2 around 400 kHz
+        - 3 # range 3 around 800 kHz
+        - 4 # range 4 around 1M Hz
+        - 5 # range 5 around 2 MHz (reset value)
+        - 6 # range 6 around 4 MHz
```
Former comment was good
         range 0 around 65.536 kHz
         range 1 around 131.072 kHz
         range 2 around 262.144 kHz
         range 3 around 524.288 kHz
         range 4 around 1.048 MHz
         range 5 around 2.097 MHz (reset value)
         range 6 around 4.194 MHz

erwango replied:
Good catch, txs

erwango replied:
Fixed

erwango commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -231,17 +218,6 @@ void config_src_sysclk_pll(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
-	 * prescaler here. In this configuration, that's the value to
-	 * use when the SYSCLK source is the PLL, not HSI.
-	 */
-	clock_switch_to_hsi(LL_RCC_SYSCLK_DIV_1);
```
Sure, will fix

FRASTM commented on drivers/clock_control/clock_stm32_ll_common.c:
```diff
@@ -253,57 +120,51 @@ static int stm32_clock_control_get_subsys_rate(const struct device *clock,
 	uint32_t ahb3_clock = get_bus_clock(ahb_clock * STM32_CPU1_PRESCALER,
 					    STM32_AHB3_PRESCALER);
+#elif defined(STM32_CLOCK_BUS_AHB3)
+	/* APB3 bus exists, but w/o dedicated prescaler */
```
/*  AHB3 bus exists

FRASTM reviewed (commented):
LGTM

erwango committed:
tests/drivers/clock_control: stm32u5: Rework to explicitly test HCLK

Instead of testing SysClockFreq setting, we should instead check HCLK
setting which is the real zephyr CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
counterpart (core clock freq) and takes AHB prescaler setting into
account.

Additionally, update one test configuration to explicitly verify AHB
prescaler is correctly taken into account by clock driver.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented on drivers/clock_control/clock_stm32f0_f3.c:
```diff
@@ -42,17 +43,40 @@ void config_pll_init(LL_UTILS_PLLInitTypeDef *pllinit)
+	} else if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
+		pll_source = LL_RCC_PLLSOURCE_HSI;
+	} else {
+		return -ENOTSUP;
```
ok, but theoretically we shouldn't get to this function if PLL is not set.
Anyway thanks for the notice, indeed we might have to come back to that part later.

erwango committed:
drivers/clock_control: stm32u5: Fix on flash latency procedure

Instead of computing hclk freq use for flash latency setting after
setting the PLLs, do it right at the beginning of the function.
Indeed, first step of PLL configuration is to switch back sysclock
to HSI source (in case it was initially PLL).
In that case, flash latency is theoretically set in consistency with PLL
driver hclk. So we should "measure" hclk freq at that step rather than
once sysclock is back on HSI.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
tests/drivers/clock_control: stm32_common: Test HCLCK instead of SYSCLK

Rework test_*_freq to test HCLK freq instead of SYSCLK one, as it is not
correct to compare CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC with SYSCLK.

Additionally, add a test to verify use of AHB prescaler.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -211,6 +169,24 @@ static struct clock_control_driver_api stm32_clock_control_api = {
 	.get_rate = stm32_clock_control_get_subsys_rate,
```
`return 0;`

ABOSTM reviewed (dismissed):
LGTM

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -386,149 +313,167 @@ void config_src_sysclk_hse(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
 
-	/* Update SystemCoreClock variable */
-	LL_SetSystemCoreClock(new_hclk_freq);
+		if (IS_ENABLED(STM32_MSIK_PLL_MODE)) {
```
according to the  [ref man](https://www.st.com/resource/en/reference_manual/rm0456-stm32u575585-armbased-32bit-mcus-stmicroelectronics.pdf)  the MSIS is the source for the PLL. However if MSIK is selected, there is a bit MSIPLLSEL in the RCC->CR to reset  with `LL_RCC_SetMSIPLLMode()`

erwango replied:
MSIPLLMode is a MSI clk calibration feature. This is different from MSI being PLL source

FRASTM commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -386,149 +313,167 @@ void config_src_sysclk_hse(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
+				"MSIS Hardware auto calibration needs LSE clock activation");
+			/* Enable MSI hardware auto calibration */
+			LL_RCC_MSI_EnablePLLMode();
+		}
```
according to the  [ref man](https://www.st.com/resource/en/reference_manual/rm0456-stm32u575585-armbased-32bit-mcus-stmicroelectronics.pdf)  the MSIS is the source for the PLL. 
Should the MSIPLLSEL in the RCC->CR be set  with `LL_RCC_SetMSIPLLMode()`  ?

erwango replied:
MSIPLLMode is a MSI clk calibration feature. This is different from MSI being PLL source

erwango committed:
include/dt-bindings: clocks: stm32h7: Use _C1_ registers offset

STM32H7 series offer alias addresses to access some registers that could
be accessed by the M4 core on dual core variants.
For instance RCC_AHB3ENR could be accessed at following offsets:
- 0x0D4: Accessible from both cores
- 0x134: Accessible from C1 (M7) core
- 0x194: Accessible from C2 (M4) core (if any)

For most single core H7 variants, the two first addresses were accessible,
but for some others (stm32h7ax/stm32h7bx), only the 'C1 accessible'
was available.

This fact used to be hidden by the use of LL API to access these registers,
providing the required abstraction (an mainly using the first alias
when possible to simplify implementation).

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

mbolivar-nordic commented:
> PR is ready for merge, but set as DNM as I will be awk next week and since there is some risk involved for such a change, I prefer it is merged when I am nearby for the after sale support.

Noted, but please notice that this needs a rebase anyway

carlescufi commented:
@nordic-krch could you please take a look?
